Mpl 1.0 compatibility

glyg opened the pull request:
Ok, PR to the correct brannch now

As I said, There ain't a correct colormap yet. Also, shall there not be a Look Up Table implemented some how (or an existing solution somewhere)?

G.

Commit:
first implementation of the ContrastSetter plugin

tonysyu committed:
ENH: Indicate frame number in collection viewer.

Commit:
Working on the contrastsetter plugin. Initial commit

	new file:   contrastsetter_demo.py
	new file:   ../../skloupe/plugins/contrastsetter.py

Commit:
Merge branch 'master' into mpl-1.0-compatibility

Commit:
First working version of the Contrast Setter plugin.

There still is an issue with the colormap not updating correctly in
the image window, but the interactions are there at least.
	modified:   ../skloupe/plugins/contrastsetter.py

tonysyu committed:
Merge branch 'mpl-1.0-compatibility'

tonysyu committed:
ENH: Add `climits` property to ImageViewer.

tonysyu committed:
ENH: fix clipping behavior of slider handle.

tonysyu committed:
ENH: Renaming and new docstrings for clarity.

API Changes:
* `update_image` --> `update_index`
* `set_image` --> `update_image`

tonysyu committed:
ENH: Add `update` parameter to `CollectionViewer`.

tonysyu commented:
This is looking really good!

I'll post some in-line comments later, but first some general comments:
- I just added a property to ImageViewer to control color limits in [a recent commit](https://github.com/tonysyu/skloupe/commit/1e793656905d5b196382350a27aa1864a875f738). If you merge in master, then you can adjust the ImageViewer's `climit` when the sliders are adjusted so that colors are properly scaled.
- It sounds like you're interested in implementing a general-purpose exposure tool (based on your comments about gamma correction). If you decide to go that route, I'd rename to `ImageAdjust` rather than `ContrastSetter`.

Minor comments (or "sorry for being obsessive-compulsive";):
- Could you stick to a single blank line between methods and between lines in the same method ([PEP8](http://www.python.org/dev/peps/pep-0008/#blank-lines)).
- If you can, remove unnecessary whitespace (whitespace at the end of lines). Most editors have plugins for doing this automatically---If not, don't worry about it; I can take care of it later.

tonysyu commented on examples/plugins/contrastsetter_demo.py:
```diff
@@ -0,0 +1,10 @@
+from skloupe.plugins import ContrastSetter
+
+
+image = data.coins()#skimage.img_as_float(data.coins())
```
You can remove the commented line and the unused `skimage` import

tonysyu commented on skloupe/plugins/contrastsetter.py:
```diff
@@ -0,0 +1,144 @@
+import matplotlib.pyplot as plt
+from skimage.filter import canny
```
unused import

tonysyu commented on skloupe/plugins/contrastsetter.py:
```diff
@@ -0,0 +1,144 @@
+        clip = low_value, high_value
+
+        hist_lines = ax_histo.step(self.bin_centers, self.hist, 
+                                   color = 'k', alpha = 1.)
```
maybe `color = 'r'` so it shows up clearly on gradient background.

tonysyu commented on skloupe/plugins/contrastsetter.py:
```diff
@@ -0,0 +1,144 @@
+        if len(self.ax_histo.images) > 0 :
+            del self.ax_histo.images[-1]
+        self.ax_histo.imshow(self.colorbar, aspect = 'auto',
+                             extent = extent)
```
Maybe add a black rectangle to the colorbar plot so it's clear that values below the low threshold get clipped to black?

tonysyu commented on skloupe/plugins/contrastsetter.py:
```diff
@@ -0,0 +1,144 @@
+                                  value=high_value, on_release=self.update_image)
+        self.slider_low = Slider(ax_low, clip, label='Minimum',
+                                  value=low_value, on_release=self.update_image)
+
```
Could you set `slidermin` and `slidermax` to prevent high < low. (see [edge detector plugin](https://github.com/tonysyu/skloupe/blob/master/skloupe/plugins/edgedetector.py) for details)

glyg commented:
Le 20/03/2012 22:42, tonysyu a écrit :

> This is looking really good!
> Thx :)
> 
> I'll post some in-line comments later, but first some general comments:
> - I just added a property to ImageViewer to control color limits in [a recent commit](https://github.com/tonysyu/skloupe/commit/1e793656905d5b196382350a27aa1864a875f738). If you merge in master, then you can adjust the ImageViewer's `climit` when the sliders are adjusted so that colors are properly scaled.
>   Ok, nice
> - It sounds like you're interested in implementing a general-purpose exposure tool (based on your comments about gamma correction). If you decide to go that route, I'd rename to `ImageAdjust` rather than `ContrastSetter`.
>   Well, I think it should be a separate plugin, as I think it is important 
>   to differentiate linear contrast adjustment to non linear LUT 
>   modifications (e.g. because some research journal accept the first one 
>   and reject the others). Right now I have to work on some actual image 
>   analysis, but I'll try to look at this later on.
>   Minor comments (or "sorry for being obsessive-compulsive";):
>   That's ok, I tend not to be careful enough, so that's a way for me to 
>   learn good practices. I'll try to clean all this up.
> - Could you stick to a single blank line between methods and between lines in the same method ([PEP8](http://www.python.org/dev/peps/pep-0008/#blank-lines)).
> - If you can, remove unnecessary whitespace (whitespace at the end of lines). Most editors have plugins for doing this automatically---If not, don't worry about it; I can take care of it later.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/tonysyu/skloupe/pull/3#issuecomment-4606054

Commit:
Code cleaning - trying to be more PEP8 complient
	modified:   plugins/contrastsetter_demo.py
	modified:   ../skloupe/plugins/contrastsetter.py

Commit:
Merge branch 'master' of github.com:tonysyu/skloupe into mpl-1.0-compatibility

Commit:
Modified update_image to use the ``imageview.climit()`` method instead
of ``image.clip()``
	modified:   ../skloupe/plugins/contrastsetter.py
